nixos/modules/services/audio/jmusicbot.nix

```diff
@@ -0,0 +1,42 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+let
+  cfg = config.services.jmusicbot;
+in
+{
+  options = {
+    services.jmusicbot = {
+      enable = mkEnableOption "jmusicbot, a Discord music bot that's easy to set up and run yourself!";
+
+      serversettings = mkOption {
+        type = types.path;
+        description = ''
+          Location where config.txt and serversettings.json is saved.
+        '';
+        default = "/var/lib/jmusicbot/";
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    system.activationScripts.jmusicbot = ''
```

Comment:
Ideally you would use `StateDirectory` to provision this. See https://github.com/NixOS/nixpkgs/pull/106697 for an example.

Alternatively `systemd.tmpfiles.rules`.